dace/libraries/standard/nodes/gearbox.py

```diff
@@ -16,73 +16,240 @@ def expansion(node: "Gearbox", parent_state: dace.SDFGState,
+                                      transient=True)
+                pack_access = write_state.add_write(pack_name)
+                unroll_entry, unroll_exit = write_state.add_map(
+                    f"{node.name}_elementwise", {elem_it: f"0:{small_veclen}"},
+                    schedule=node.schedule)
+                unroll_tasklet = write_state.add_tasklet(
+                    f"{node.name}_elementwise", {"buffer_in"}, {"pack_out"},
+                    "pack_out = buffer_in")
+                write_state.add_memlet_path(
+                    buffer_read,
+                    unroll_entry,
+                    unroll_tasklet,
+                    dst_conn="buffer_in",
+                    memlet=dace.Memlet(f"{buffer_name_inner}[{vec_it} * "
+                                       f"{small_veclen} + {elem_it}]"))
+                write_state.add_memlet_path(
+                    unroll_tasklet,
+                    unroll_exit,
+                    pack_access,
+                    src_conn="pack_out",
+                    memlet=dace.Memlet(f"{pack_name}[{elem_it}]"))
+                write_state.add_memlet_path(
+                    pack_access,
+                    write_nested,
+                    memlet=dace.Memlet(f"{write_nested.data}[0]"))
+
+            nested_sdfg_node = state.add_nested_sdfg(nested_sdfg, sdfg,
+                                                     {f"_{in_edge.dst_conn}", f"{buffer_name}_inner"},
+                                                     {f"_{out_edge.src_conn}", f"{buffer_name}_inner"})
+            buffer_read = state.add_read(buffer_name)
+            buffer_write = state.add_write(buffer_name)
+            state.add_memlet_path(input_read,
+                                  entry,
+                                  nested_sdfg_node,
+                                  dst_conn=f"_{in_edge.dst_conn}",
+                                  memlet=dace.Memlet(f"{input_read.data}[0]",
+                                                     dynamic=True))
+            state.add_memlet_path(nested_sdfg_node,
+                                  exit,
+                                  output_write,
+                                  src_conn=f"_{out_edge.src_conn}",
+                                  memlet=dace.Memlet(f"{output_write.data}[0]",
+                                                     dynamic=True))
+            state.add_memlet_path(
+                buffer_read,
+                entry,
+                nested_sdfg_node,
+                dst_conn=buffer_name_inner,
+                memlet=dace.Memlet(f"{buffer_name}[0:{large_veclen}]"))
+            state.add_memlet_path(
+                nested_sdfg_node,
+                exit,
+                buffer_write,
+                src_conn=buffer_name_inner,
+                memlet=dace.Memlet(f"{buffer_name}[0:{large_veclen}]"))
+
+        else:  # Not elementwise, one side is a vector of vectors
```

Comment:
Do we have cases where this actually happens?

Reply:
It's two different styles of programming it -- honestly the vector of vectors is a bit more representative of what we do (since it enforces that it's a multiple). The problem is that it doesn't work with Intel, which is why I needed to implement the other version also...